parastell/source_mesh.py

```diff
@@ -133,18 +134,54 @@ def __init__(
 
         self._create_mbc()
 
+    @property
+    def num_poloidal_pts(self):
+        return self._num_poloidal_pts
+
+    @num_poloidal_pts.setter
+    def num_poloidal_pts(self, value):
+        if value % 2 != 1:
+            e = AttributeError(
+                "To ensure that tetrahedral faces are coincident at the end of "
+                "the closed poloidal loop, the number of poloidal intervals "
+                "must be even. To ensure this, the number of poloidal grid "
+                "points must be odd."
+            )
+            self._logger.error(e.args[0])
+            raise e
+        self._num_poloidal_pts = value
+
+    @property
+    def num_toroidal_pts(self):
+        return self._num_toroidal_pts
+
+    @num_toroidal_pts.setter
+    def num_toroidal_pts(self, value):
+        self._num_toroidal_pts = value
+
     @property
     def toroidal_extent(self):
         return self._toroidal_extent
 
     @toroidal_extent.setter
     def toroidal_extent(self, angle):
-        self._toroidal_extent = np.deg2rad(angle)
-        if self._toroidal_extent > 360.0:
-            e = ValueError("Toroidal extent cannot exceed 360.0 degrees.")
+        if angle > 360.0:
+            e = AttributeError("Toroidal extent cannot exceed 360.0 degrees.")
+            self._logger.error(e.args[0])
+            raise e
+
+        if angle == 360.0 and self._num_toroidal_pts % 2 != 1:
```

Comment:
Later on you compare `_toroidal_extent` to 2 pi.  It might make more sense to convert to radians before this test for consistency.